wallet-new/server/Cardano/Wallet/API/V1/Handlers/Addresses.hs

```diff
@@ -31,3 +33,18 @@ listAddresses PaginationParams {..} =
 
 newAddress :: Address -> Handler Address
 newAddress a = return a
+
+-- | Verifies that an address is < 78 and base58 decodable.
+verifyAddress :: Text -> Handler AddressValidity
+verifyAddress address
+  | length address > maxAddressSize =
+      return $ AddressValidity False
+  | otherwise =
+      case decodeTextAddress address of
```

Comment:
You can skip the double `return $` if you wrap the `case` block into `return` this way:

``` haskell
return $ case foo of
  Left _ -> True
  Right _ -> False
```